docs/Advanced/Tasks Api.md

```diff
@@ -32,6 +32,16 @@ export interface TasksApiV1 {
      * an empty string, if data entry was cancelled.
      */
     createTaskLineModal(): Promise<string>;
+
+    /**
+     * Toggles and updates a task line according to a user's preferences, accounting for recurrance
+     * rules and completed status.
+     *
+     * @param line The markdown string of the task line being toggled
+     * @param path The path to the file containing line
+     * @returns An {@link EditorInsertion} containing the information necessary to replace the toggled line
+     */
+    toggleLine: (line: string, path: string) => EditorInsertion;
```

Comment:
Could you please rename this to `executeToggleTaskDoneCommand()` so that this code is self documenting. There is a lot more behaviour inside than just toggling, so this makes the name honest =) and it leaves us free to add more toggle overloads in future with more precise programmatic control.

We are asking as @claremacrae and myself have to think about broader use cases for other plugins as well.

Reply:
Thanks @ilandikov. What would you think about shortening it a little to `executeToggleTaskCommand()` since this function handles both checking and unchecking a task?

Reply:
Actually that name was chosen to be self documenting with the [toggle task done](https://publish.obsidian.md/tasks/Editing/Toggling+and+Editing+Statuses#'Tasks+Toggle+task+done'+command) command =)

Reply:
Looking at that link to the docs, it would be worth adding a row the table in that section to indicate that the command is now available via the API.

Reply:
@claremacrae I've approved this PR with the latest changes, I don't think a row would be good for the table since it has a 'Where' and 'Viewing mode' columns that make little sense for an API call AFAIK. Please let me know if the current change is ok.

Reply:
I do like the new wording, and take the point about the table not being the right location...

Remember to add the version number info - we have templates for that:
https://publish.obsidian.md/tasks-contributing/Documentation/Version+numbers+in+documentation